lib/logstash-logger/device/firehose.rb

```diff
@@ -0,0 +1,41 @@
+require 'aws-sdk'
+require 'logstash-logger/device/aws_stream'
+
+module LogStashLogger
+  module Device
+    class Firehose < AwsStream
+
+      def initialize(opts)
+        super
+        @stream_class = ::Aws::Firehose::Client
+        @recoverable_error_codes = [
```

Comment:
Would be better if this could be a class instance variable. It's class level configuration that never changes so it shouldn't be reinitialized with every new object.